clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/NodeStateChangeChecker.java

```diff
@@ -354,7 +398,22 @@ private static boolean groupContainsNode(Group group, Node node) {
         return false;
     }
 
+    private static boolean oneOfGroupsContainsNode(Collection<Group> groups, Node node) {
+        for (Group group : groups) {
+            for (ConfiguredNode configuredNode : group.getNodes()) {
+                if (configuredNode.index() == node.getIndex()) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
     private Result checkAllNodesAreUp(ClusterState clusterState) {
+        // TODO Revisit this, check that all nodes in other groups are up?
+        if (moreThanOneGroupAllowedToBeDown()) return allowSettingOfWantedState();
```

Comment:
I think we need to ponder and discuss the desired semantics for this one a bit, probably when @hakonhall is back

Reply:
Not sure If I understand how more than one node can be set to maintenance (as is the goal when allowing more than one group to be down at a time) if that is the case? This methods checks both wanted state and actual state, so name is a bit misleading.

Reply:
Discussed offline